### PR TITLE
[20318] Fix and refactor Windows Github CI

### PR DESCRIPTION
**Pull request by EduPonz**
## Description

This PR:
* Fixes Windows CI in Github (this has implied disabling PKCS#11 tests for now; reenabling them will be investigated in a different PR)
* Refactors the workflow to use more of eProsima-CI features
* Builds with warnings as errors
* Uses Asio and TinyXML2 from thirdparty instead of choco installation because the post build steps of the test related CMake targets is not being able to find one of them (I assume TinyXML2) when running the test executable to discover the tests

@Mergifyio backport 2.12.x 2.11.x 2.10.x 2.6.x

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- _N/A_: The code follows the style guidelines of this project. 
- _N/A_: Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- _N/A_: Any new/modified methods have been properly documented using Doxygen. 
- _N/A_: Changes are ABI compatible. 
- _N/A_: Changes are API compatible. 
- _N/A_: New feature has been added to the `versions.md` file (if applicable).
- _N/A_: New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [x] Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by richiware — APPROVED** [review]

File: .github/workflows/config/default_ci.meta [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
+            - "-DPROFILING_TESTS=OFF"
+            - "-DSECURITY=ON"
+            - "-DSHM_TRANSPORT_DEFAULT=ON"
+            - "-DSTRICT_REALTIME=OFF"
+            - "-DSYSTEM_TESTS=ON"
+            - "-DTHIRDPARTY_Asio=FORCE"
+            - "-DTHIRDPARTY_fastcdr=OFF"
+            - "-DTHIRDPARTY_TinyXML2=FORCE"
+            - "-DTHIRDPARTY_UPDATE=ON"
+            - "-DEPROSIMA_EXTRA_CMAKE_CXX_FLAGS=/MP /WX"
```
Comment (richiware) [inline .github/workflows/config/default_ci.meta:26]: Not following the order

File: .github/workflows/config/nightly.meta [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
+            - "-DPERFORMANCE_TESTS=ON"
+            - "-DPROFILING_TESTS=OFF"
+            - "-DSHM_TRANSPORT_DEFAULT=ON"
+            - "-DSTRICT_REALTIME=OFF"
+            - "-DSYSTEM_TESTS=ON"
+            - "-DTHIRDPARTY_Asio=FORCE"
+            - "-DTHIRDPARTY_fastcdr=OFF"
+            - "-DTHIRDPARTY_TinyXML2=FORCE"
+            - "-DTHIRDPARTY_UPDATE=ON"
+            - "-DEPROSIMA_EXTRA_CMAKE_CXX_FLAGS=/MP /WX"
```
Comment (richiware) [inline .github/workflows/config/nightly.meta:25]: Not following the order

File: .github/workflows/reusable-windows-ci.yml [hunk 3]
```diff
@@ -153,14 +63,20 @@ jobs:
           if (Test-Path -Path $Env:ProgramW6432\OpenSSL)
           {
             "OPENSSL64_ROOT=$Env:ProgramW6432\OpenSSL" | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
+            Set-Item -Force -Path "Env:PATH" -Value "$Env:PATH;$Env:ProgramW6432\OpenSSL"
```
Comment (richiware) [inline .github/workflows/reusable-windows-ci.yml:66]:
```suggestion
            Set-Item -Force -Path "Env:PATH" -Value "$Env:PATH;$Env:ProgramW6432\OpenSSL\bin"
```

File: .github/workflows/reusable-windows-ci.yml [hunk 4]
```diff
@@ -153,14 +63,20 @@ jobs:
...
+            "PATH=$Env:PATH" | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
           }
           elseif (Test-Path -Path $Env:ProgramW6432\OpenSSL-Win)
           {
             "OPENSSL64_ROOT=$Env:ProgramW6432\OpenSSL-Win" | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
+            Set-Item -Force -Path "Env:PATH" -Value "$Env:PATH;$Env:ProgramW6432\OpenSSL-Win"
```
Comment (richiware) [inline .github/workflows/reusable-windows-ci.yml:72]:
```suggestion
            Set-Item -Force -Path "Env:PATH" -Value "$Env:PATH;$Env:ProgramW6432\OpenSSL-Win\bin"
```

File: .github/workflows/reusable-windows-ci.yml [hunk 5]
```diff
@@ -153,14 +63,20 @@ jobs:
...
+            "PATH=$Env:PATH" | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
           }
           elseif (Test-Path -Path $Env:ProgramW6432\OpenSSL-Win64)
           {
             "OPENSSL64_ROOT=$Env:ProgramW6432\OpenSSL-Win64" | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
+            Set-Item -Force -Path "Env:PATH" -Value "$Env:PATH;$Env:ProgramW6432\OpenSSL-Win64"
```
Comment (richiware) [inline .github/workflows/reusable-windows-ci.yml:78]:
```suggestion
            Set-Item -Force -Path "Env:PATH" -Value "$Env:PATH;$Env:ProgramW6432\OpenSSL-Win64\bin"
```

**Review by richiware — CHANGES_REQUESTED** [review]

File: .github/workflows/config/default_ci.meta (hunk 1 above)
Comment (EduPonz) [reply .github/workflows/config/default_ci.meta]: d6206f7

File: .github/workflows/config/nightly.meta (hunk 2 above)
Comment (EduPonz) [reply .github/workflows/config/nightly.meta]: d6206f7

File: .github/workflows/reusable-windows-ci.yml (hunk 3 above)
Comment (EduPonz) [reply .github/workflows/reusable-windows-ci.yml]: I dropped the commit

File: .github/workflows/reusable-windows-ci.yml (hunk 4 above)
Comment (EduPonz) [reply .github/workflows/reusable-windows-ci.yml]: I dropped the commit

File: .github/workflows/reusable-windows-ci.yml (hunk 5 above)
Comment (EduPonz) [reply .github/workflows/reusable-windows-ci.yml]: I dropped the commit

**Review by richiware — APPROVED** [review]

**Review by JesusPoderoso — APPROVED** [review]

**Review by richiware — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@Mergifyio backport 2.12.x 2.11.x 2.10.x 2.6.x

**Pull request merged**